File: crates/rr-core/tests/fixtures/php/surface.php
<?php

namespace App;

class Service implements Contract
{
    public const MAX = 3;
    private int $secret;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    public function run(): void
    {
        helper();
    }

    private function hidden(): void
    {
    }
}

interface Contract
{
    function run(): void;
}

function bare(string $name): string
{
    return $name;
}

File: crates/rr-core/tests/fixtures/cpp/surface.cpp
#include <string>
#include "local.h"

namespace app {
class Service {
public:
    Service();
    void run();

private:
    int secret;
};

void Service::run() {}

struct Point {
    int x;
    int y;
};

template <typename T>
T identity(T value) {
    return value;
}
}
